_pydevd_bundle/pydevd_frame.py

```diff
@@ -674,16 +698,13 @@ def trace_dispatch(self, frame, event, arg):
                 else:
                     stop = False
 
-                if stop and step_cmd != -1 and IS_PY3K:
-                    # in Py3k we start script via our custom `execfile` function, and we shouldn't stop there
-                    # while stepping when execution is finished
-                    if is_return:
-                        f_code = getattr(frame.f_back, 'f_code', None)
-                        if f_code is not None:
-                            back_filename = os.path.basename(f_code.co_filename)
-                            file_type = get_file_type(back_filename)
-                            if file_type == PYDEV_FILE:
-                                stop = False
+                if step_cmd != -1 and is_return and IS_PY3K and hasattr(frame, "f_back"):
+                    f_code = getattr(frame.f_back, 'f_code', None)
```

Comment:
I removed `stop` accidentally, you're right. But we need to check whether a `frame` object has a `f_back`. It isn't always true, so we need to check it.